### PR TITLE
Ugly Numbers (Using DP Algorithmic Approach)

### LINKED ISSUE #963: Ugly Numbers (Using DP algorithmic approach)
<hr>

## Description 📜

Problem - Given a number n, the task is to find n’th Ugly number.
I will provide a proper C++ solution regarding this problem. [DOCUMENTATION]
<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

### PR DESCRIPTION
**Pull request by swatigupta121**
<hr>

## Description 📜

I have added a well commented code for solving Ugly Numbers using dynamic approach. I also specified the output, time complexity and space complexity at the end of program. 
Fixes #963

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials) 

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Review by vasu-1 — CHANGES_REQUESTED** [review]
Good job @swatigupta121 
Please use markdown instead of cpp documentation !!

**Comment by swatigupta121** [comment]
@vasu-1 
I have performed the required changes. Please review it once again.

**Comment by vasu-1** [comment]
Hey @swatigupta121 
I am not saying like this..
Please see this md file[Link](https://github.com/girlscript/winter-of-contributing/blob/C_CPP/C_CPP/Data%20Structures/Linked%20List/Singly_Linked_List_In_C.md) 

Change the code accordingly..

**Comment by swatigupta121** [comment]
Hello @vasu-1 
I tried making the changes. Please check it once. And sorry for the inconvenience caused as I am new to open source and still exploring it.

**Review by vasu-1 — APPROVED** [review]
Nicee @su5

**Comment by vasu-1** [comment]
@Suvraneel  review and merge .

File: C_CPP/Algorithmic Approaches/Dynamic/readme.md [hunk 1]
```diff
@@ -0,0 +1 @@
+- Ugly Numbers [https://github.com/swatigupta121/winter-of-contributing/blob/swatigupta121-patch-1/C_CPP/Algorithmic%20Approaches/Dynamic/Ugly_Number.md]
```
Comment (Suvraneel) [inline C_CPP/Algorithmic Approaches/Dynamic/readme.md:1]: Use relative link...
From gwoc repo...
Not your fork

**Review by Suvraneel — CHANGES_REQUESTED** [review]

**Review by vasu-1 — APPROVED** [review]
@Suvraneel  👉👈I think we can merge this !

**Comment by Suvraneel** [comment]
@swatigupta121  Awesome ❤️  LGTM :octocat:  💯

**Pull request merged**